src/main/groovy/org/shipkit/gradle/ReleaseNeededTask.java

```diff
@@ -24,23 +19,15 @@
  *  - the current Git branch does not match release-eligibility regex ({@link #getReleasableBranchRegex()}.
  *  - binaries have not changes since the previous release
  */
+//TODO move to 'release' subpackage
 public class ReleaseNeededTask extends DefaultTask {
```

Comment:
A good candidate for being [@CacheableTask])https://docs.gradle.org/4.0/javadoc/org/gradle/api/tasks/CacheableTask.html) once only newer Gradle 3.x versions are supported.